### PR TITLE
feat(BrowserFetcher): support downloading content_shell

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch adds a "product" option to BrowserFetcher that can be
either 'chromium' or 'content_shell'.

ContentShell might be useful as a stripped-down version of Chromium.
> **NOTE**: Puppeteer API is **not** guaranteed to work with content_shell; some of the methods don't actually work. Use on your own risk.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -283,11 +283,16 @@ This methods attaches Puppeteer to an existing Chromium instance.
...
 #### puppeteer.createBrowserFetcher([options])
 - `options` <[Object]>
+  - `product` <[string]> Possible values are: `'chromium'`, `'content_shell'`. Defaults to `'chromium'`.
   - `host` <[string]> A download host to be used. Defaults to `https://storage.googleapis.com`.
   - `path` <[string]> A path for the downloads folder. Defaults to `<root>/.local-chromium`, where `<root>` is puppeteer's package root.
-  - `platform` <[string]> Possible values are: `mac`, `win32`, `win64`, `linux`. Defaults to the current platform.
+  - `platform` <[string]> Possible values are: `'mac'`, `'win32'`, `'win64'`, `'linux'`. Defaults to the current platform.
 - returns: <[BrowserFetcher]>
 
+> **NOTE** ` 'content_shell'` is a basic browser built atop of `content/` layer as a part of chromium project. Puppeteer API is not guaranteed to work with content_shell; **use on your own risk**.
```
Comment (JoelEinbinder) [inline docs/api.md]: Chromium

File: docs/api.md [hunk 2]
```diff
@@ -283,11 +283,16 @@ This methods attaches Puppeteer to an existing Chromium instance.
...
+>
+> More about `content/` layer here: [content module](https://www.chromium.org/developers/content-module).
```
Comment (JoelEinbinder) [inline docs/api.md]: The content layer

**Review by JoelEinbinder — APPROVED** [review]

File: docs/api.md (hunk 2 above)
Comment (aslushnikov) [reply docs/api.md]: Done

File: docs/api.md (hunk 1 above)
Comment (aslushnikov) [reply docs/api.md]: Done

File: docs/api.md [hunk 3]
```diff
@@ -283,11 +283,16 @@ This methods attaches Puppeteer to an existing Chromium instance.
...
 #### puppeteer.createBrowserFetcher([options])
 - `options` <[Object]>
+  - `product` <[string]> Possible values are: `'chromium'`, `'content_shell'`. Defaults to `'chromium'`.
   - `host` <[string]> A download host to be used. Defaults to `https://storage.googleapis.com`.
   - `path` <[string]> A path for the downloads folder. Defaults to `<root>/.local-chromium`, where `<root>` is puppeteer's package root.
-  - `platform` <[string]> Possible values are: `mac`, `win32`, `win64`, `linux`. Defaults to the current platform.
+  - `platform` <[string]> Possible values are: `'mac'`, `'win32'`, `'win64'`, `'linux'`. Defaults to the current platform.
 - returns: <[BrowserFetcher]>
 
+> **NOTE** ` 'content_shell'` is a basic browser built atop of `content/` layer as a part of Chromium project. Puppeteer API is not guaranteed to work with content_shell; **use on your own risk**.
```
Comment (JoelEinbinder) [inline docs/api.md]: The puppeteer api. Use at your own risk

**Comment by aslushnikov** [comment]
It turned out content_shell doesn't pass a lot of pptr tests:
- `browser.close()` is not implemented in `content/`; instead, the implementations are in `chrome/` and `headless/` layers
- javascript dialog support is not implemented in `content/` as well
- most of screenshot tests fail with opacity and other issues

There's no intent to invest into `content_shell` to fix these incompatibilities, so closing this patch. It's nice we've explored this.

**Pull request closed without merging**